### PR TITLE
docs(framework) Update how to aggregate evaluation results documentation

### PR DESCRIPTION
**Pull request by dstripelis**
- Add how to create a server function with context.
- Add how to create a server app and pass server function.
- Adding float casting on returned losses.

### CODE REVIEW

File: doc/source/how-to-aggregate-evaluation-results.rst [hunk 1]
```diff
@@ -69,11 +69,24 @@ dictionaries:
             )
 
             # Return aggregated loss and metrics (i.e., aggregated accuracy)
-            return aggregated_loss, {"accuracy": aggregated_accuracy}
+            return float(aggregated_loss), {"accuracy": aggregated_accuracy}
```
[the remark this answers is not in the document]
Comment (dstripelis) [reply doc/source/how-to-aggregate-evaluation-results.rst:72]: Fixed.

File: doc/source/how-to-aggregate-evaluation-results.rst [hunk 2]
```diff
@@ -69,11 +69,24 @@ dictionaries:
...
-    # Create strategy and run server
-    strategy = AggregateCustomMetricStrategy(
-        # (same arguments as FedAvg here)
+    def server_fn(context: Context) -> ServerAppComponents:
+        # Read federation rounds from config
+        num_rounds = context.run_config["num-server-rounds"]
+        config = ServerConfig(num_rounds=num_rounds)
+
+        # Define strategy
+        strategy = AggregateCustomMetricStrategy()
```
[the remark this answers is not in the document]
Comment (dstripelis) [reply doc/source/how-to-aggregate-evaluation-results.rst:81]: Fixed.

**Comment by dstripelis** [comment]
@jafermarq thank you for your feedback! In order to have some isolation between PRs, in the current PR I only improve the custom strategy class and address @chongshenng's comments.

However, following your suggestions, I created a new PR where I introduce the evaluation callback function in the documentation: https://github.com/adap/flower/pull/4544, while keeping the custom class strategy. I think we can review and include this PR in the next release.

**Comment by chongshenng** [comment]
@dstripelis @jafermarq  I agree, let's focus on updating the docs for the 1.13 release, then in a subsequent PR we introduce the callback too.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**